test/source/TestUtility.cpp

```diff
@@ -543,6 +554,20 @@ static int TestUtilityExchange()
 		EATEST_VERIFY(a == "EA Standard Template Library");
 	}
 
+	// Issue #82; construct pair using single move constructor
+	#if EASTL_MOVE_SEMANTICS_ENABLED
+	{
+		int i1 = 1;
+		TestPairSingleMoveConstructor<int> test1;
+		test1.test(eastl::move(i1));
+
+		int i2a = 2;
+		int & i2b = i2a;
+		TestPairSingleMoveConstructor<int&> test2;
+		test1.test(eastl::move(i2b));
```

Comment:
Ok I think I have a new candidate:
```
			template <typename U, typename = typename eastl::enable_if<eastl::is_convertible<U&&,typename eastl::remove_reference<T1>::type&&>::value>::type>
			EA_CPP14_CONSTEXPR pair(U&& x)
				: first(eastl::move(x)),
					second()
			{
			}
```

The problem is I don't think this is a useful test:
`test2.test(eastl::move(i2b))`

Because move basically does this:
`static_cast<typename eastl::remove_reference<decltype(x)>::type&&>(x)`

The critical part being the remove_reference<> applied to reference generates a temporary int... which results in this error:
```
EASTL/test/../include/EASTL/utility.h:420:7: error: non-const lvalue reference to type 'int' cannot bind to a temporary of type 'typename eastl::remove_reference<int &>::type'
      (aka 'int')
                                : first(eastl::move(x)),
                                  ^     ~~~~~~~~~~~~~~
EASTL/test/source/TestUtility.cpp:31:24: note: in instantiation of function template specialization 'eastl::pair<int &, int>::pair<int, void>' requested here
                                eastl::pair<T1,T2> p(eastl::move(t1));
                                                   ^
EASTL/test/source/TestUtility.cpp:578:10: note: in instantiation of member function 'TestPairSingleMoveConstructor<int &, int>::test2' requested here
                        test2.test2(i2b);
                              ^
```